webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/campaigns/MainCampaignService.scala

```diff
@@ -225,33 +225,36 @@ class MainCampaignService(repo: CampaignEventRepository, campaignRepo: CampaignR
     def start() = loop().forever
   }
 
-  def nextCampaignDate(schedule : CampaignSchedule, date : DateTime) : IOResult[DateTime] = {
+  def nextDateFromDayTime(date : DateTime, start : DayTime) : DateTime = {
```

Comment:
this will need comments and unit tests, I spent quite some time understanding the logic here (if I got it: "date" is before; start, increment by one week but get all other time info from "start". I'm not sure about what happen if date is several weeks behind start, perhaps it should not happen ?)